### PR TITLE
#300 Allow Invoice Payment Only If totalAmount >= 108 EUR

### LINKED ISSUE #300: Allow Invoice Payment Only If totalAmount >= 108 EUR
We made this change in the core, but we also have to do it on the web UI.

### PR DESCRIPTION
**Pull request by criske**
fixes #300

Also active invoices with 0 total amount can't be downloaded. (I think this bug was reintroduced after we moved the pdf generation logic on server)

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by criske** [comment]
no bounty

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2021-01-07T19:38:53.822733``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

**Comment by criske** [comment]
@amihaiemil speaking of pdf invoices, shouldn't we delete them from server after user finished the download?

File: src/main/resources/public/js/getAndAddContracts.js [hunk 1]
```diff
@@ -157,13 +157,17 @@ var projectContractsCount = -1;
                 + "</a>  ";
             if (invoice.paymentTime == "null" && invoice.transactionId == "null") {
                 status = "Active";
-                if (parseFloat(invoice.totalAmount) > 0.0) {
+                var totalAmount = parseFloat(invoice.totalAmount.substring(0, invoice.totalAmount.length - 1))
```
Comment (criske) [inline src/main/resources/public/js/getAndAddContracts.js:160]: I think I have a bug here though, total amount has comma as decimal separator and messes the parseFloat.

**Comment by amihaiemil** [comment]
@criske  > Also active invoices with 0 total amount can't be downloaded.

Please change this pack, I want to allow invoice download even if the value is 0 :D 
This way, the client and/or the contributor can see early if there is any problem with the Billing Info they entered.

Yes, we could remove the invoices after download, but I'm not sure how to do that in the Spring RestController.

**Comment by criske** [comment]
> Please change this pack, I want to allow invoice download even if the value is 0 :D
> This way, the client and/or the contributor can see early if there is any problem with the Billing Info they entered.

Right, that makes sense.

> Yes, we could remove the invoices after download, but I'm not sure how to do that in the Spring RestController.

The easiest way is to use `StreamingResponseBody`, I think.  When we're closing the stream in `StreamingResponseBody#writeTo(OutputStream)` we delete the file too.

**Comment by amihaiemil** [comment]
@criske 

> The easiest way is to use StreamingResponseBody, I think. When we're closing the stream in StreamingResponseBody#writeTo(OutputStream) we delete the file too.

Open an Issue about this and I'll assign it to you :D

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**